slo-workload/src/Commands/RunCommand.php

```diff
@@ -0,0 +1,421 @@
+        } catch (\Exception $e) {
+            echo $e->getMessage();
+        }
+
+        while (microtime(true) <= $startTime + $time) {
+            $begin = microtime(true);
+            Utils::metricInflight("read", $this->queueId);
+            $attemps = 0;
+            try {
+                $table->retryTransaction(function (\YdbPlatform\Ydb\Session $session)
+                use ($query, $dataGenerator, $tableName, &$attemps) {
+                    try {
+                        $attemps++;
+                        return $session->query($query, [
+                            "\$id" => (new \YdbPlatform\Ydb\Types\Uint64Type($dataGenerator->getRandomId()))->toTypedValue()
+                        ]);
+                    } catch (\Exception $exception) {
+                        Utils::retriedError($this->queueId, 'read', get_class($exception));
+                    }
+                }, true, null, [
+                    'callback_on_error' => function (\Exception $e) {
+                        Utils::retriedError($this->queueId, 'write', get_class($e));
+                    }
+                ]);
+                Utils::metricDone("read", $this->queueId, $attemps, (microtime(true) - $begin) * 1000);
+            } catch (\Exception $e) {
+                if ($attemps == 0) $attemps++;
+                $table->getLogger()->error($e->getMessage());
+                Utils::metricFail("read", $this->queueId, $attemps, get_class($e), (microtime(true) - $begin) * 1000);
+            } finally {
+                $delay = ($begin - microtime(true)) * 1e6 + 1e6 / Defaults::RPS_PER_WRITE_FORK;
+                usleep($delay > 0 ? $delay : 1);
+            }
+
+        }
+
+    }
+
+    protected function writeJob(string $endpoint, string $path, $tableName, int $initialDataCount,
+                                int    $time, int $writeTimeout, int $process, int $shutdownTime, int $startTime)
+    {
+        try {
+            $ydb = Utils::initDriver($endpoint, $path, "write-$process");
+            $dataGenerator = new DataGenerator();
+            $dataGenerator::setMaxId($initialDataCount);
+            $query = sprintf(Defaults::WRITE_QUERY, $tableName);
+            $table = $ydb->table();
+        } catch (\Exception $e) {
+            echo $e->getMessage();
+        }
+
+        while (microtime(true) <= $startTime + $time) {
+            $begin = microtime(true);
+            Utils::metricInflight("write", $this->queueId);
+            $attemps = 0;
+            try {
+                $table->retryTransaction(function (\YdbPlatform\Ydb\Session $session)
+                use ($query, $dataGenerator, $tableName, &$attemps) {
+                    $attemps++;
+                    return $session->query($query, DataGenerator::getUpsertData());
+                }, true, null, [
+                    'callback_on_error' => function (\Exception $e) {
+                        Utils::retriedError($this->queueId, 'write', get_class($e));
```

Comment:
What about send log messages to main process by queue and write it to stderr too?